### PR TITLE
HPCC-12414 Row tag required for spray xml

### PR DESCRIPTION
**Pull request by miguelvazq**
Make a row tag be a requirement whenever spraying to xml.

Signed-off by: Miguel Vazquez miguel.vazquez@lexisnexis.com

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/eclwatch/LZBrowseWidget.js [hunk 1]
```diff
@@ -393,16 +393,25 @@ define([
...
                 });
-                FileSpray.SprayVariable({
-                    request: request
-                }).then(function (response) {
-                    context._handleResponse("SprayResponse.wuid", response);
+            } else {
+                topic.publish("hpcc/brToaster", {
+                    Severity: "Error",
+                    Source: "FileSpray.SprayVariable",
+                    Exceptions: [{ Message: "You must supply a target row tag." }]
```
Comment (GordonSmith) [inline esp/src/eclwatch/LZBrowseWidget.js]: Needs to be a i18n string.

File: esp/src/eclwatch/LZBrowseWidget.js [hunk 2]
```diff
@@ -393,16 +393,25 @@ define([
 
         _onSprayXml: function(event) {
             var context = this;
-            this._spraySelectedOneAtATime("SprayXmlDropDown", "SprayXmlForm", function (request, item) {
-                lang.mixin(request, {
-                    sourceRowTag: item.targetRowTag
+            this.sprayXmlTargetRowTag = registry.byId(this.id + "sprayXmlTargetRowTag");
+            if (this.sprayXmlTargetRowTag.getValue() !== "") {
```
Comment (GordonSmith) [inline esp/src/eclwatch/LZBrowseWidget.js]: (Minor) I would probably just have:
```
if (this.sprayXmlTargetRowTag.getValue()) {
```
That way it also tests for null as well as "" (Its minor as I don't think you can get a null / undefined here).

**Comment by richardkchapman** [comment]
@GordonSmith Please re-review

**Comment by GordonSmith** [comment]
Looks good.
